pkg/dependencies/factory.go

```diff
@@ -403,6 +407,38 @@ func (f *Factory) WithProvider(clusterConfigFile string, clusterConfig *v1alpha1
 				f.dependencies.Kubectl,
 				time.Now,
 			)
+		case v1alpha1.NutanixDatacenterKind:
+			datacenterConfig, err := v1alpha1.GetNutanixDatacenterConfig(clusterConfigFile)
+			if err != nil {
+				return fmt.Errorf("unable to get datacenter config from file %s: %v", clusterConfigFile, err)
+			}
+
+			machineConfigs, err := v1alpha1.GetNutanixMachineConfigs(clusterConfigFile)
+			if err != nil {
+				return fmt.Errorf("unable to get machine config from file %s: %v", clusterConfigFile, err)
+			}
+
+			url := fmt.Sprintf("%s:%d", datacenterConfig.Spec.Endpoint, datacenterConfig.Spec.Port)
+			nutanixCreds := prismgoclient.Credentials{
+				URL:      url,
+				Username: os.Getenv("NUTANIX_USER"),
+				Password: os.Getenv("NUTANIX_PASSWORD"),
+				Endpoint: datacenterConfig.Spec.Endpoint,
+				Port:     fmt.Sprintf("%d", datacenterConfig.Spec.Port),
+			}
+			client, err := v3.NewV3Client(nutanixCreds)
+			if err != nil {
+				return fmt.Errorf("error creating nutanix client: %v", err)
+			}
+
+			validator, err := nutanix.NewValidator(client.V3)
+			if err != nil {
+				return err
+			}
```

Comment:
any reason to initialize this here instead of from within the provider? It only needs the client which is already being passed to the provider

Reply:
Nothing in particular, can be moved into the provider.